rfcs/react-components/convergence/motion-definition-n-apis.md

```diff
@@ -0,0 +1,999 @@
+
+[@layershifter](https://github.com/layershifter)
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Summary](#summary)
+- [Background](#background)
+  - [Motion definition](#motion-definition)
+  - [Motion APIs](#motion-apis)
+  - [Motion usage & replacement](#motion-usage--replacement)
+- [Web Animations API over CSS way](#web-animations-api-over-css-way)
+  - [Performance & re-renders](#performance--re-renders)
+    - [State lifecycle for CSS option](#state-lifecycle-for-css-option)
+    - [State lifecycle for Web Animations API](#state-lifecycle-for-web-animations-api)
+    - [Sample profiles](#sample-profiles)
+  - [Web Animations are framework-agnostic](#web-animations-are-framework-agnostic)
+  - [No need to sync CSS & JS like with CSS option](#no-need-to-sync-css--js-like-with-css-option)
+  - [Potential limits in API with CSS option](#potential-limits-in-api-with-css-option)
+- [Detailed design](#detailed-design)
+  - [Motion definition](#motion-definition-1)
+  - [Using atom motions](#using-atom-motions)
+  - [Using presence motions](#using-presence-motions)
+  - [Using group motions](#using-group-motions)
+  - [Motion overrides](#motion-overrides)
+- [Discarded solution (CSS option)](#discarded-solution-css-option)
+  - [Motion definition](#motion-definition-2)
+  - [Using atom motions](#using-atom-motions-1)
+    - [Apply classes to an element](#apply-classes-to-an-element)
+    - [Factories](#factories)
+  - [Using presence motions](#using-presence-motions-1)
+    - [Factories & utility components](#factories--utility-components)
+  - [Using group motions](#using-group-motions-1)
+  - [Motion overrides](#motion-overrides-1)
+    - [Using `motion` & `className` props](#using-motion--classname-props)
+    - [Using `motion` prop as object](#using-motion-prop-as-object)
+- [Notes](#notes)
+  - [Web Animations API have higher priority in applying styles](#web-animations-api-have-higher-priority-in-applying-styles)
+- [Open questions](#open-questions)
+  - [Split definition & APIs](#split-definition--apis)
+  - [What to export from `@fluentui/react-components`?](#what-to-export-from-fluentuireact-components)
+  - [Mouse over / focus / active states](#mouse-over--focus--active-states)
+  - [Migration plan](#migration-plan)
+- [Out of scope](#out-of-scope)
+  - [Motion sequencing & grouping](#motion-sequencing--grouping)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Summary
+
+This RFC proposes the way to define motion animations and APIs to use them in Fluent UI React components (`@fluentui/react-components`).
+
+## Background
+
+There are three main use cases that this RFC is trying to solve:
+
+- **Motion definition**: How should motion animations be defined?
+- **Motion APIs**: Which APIs should be exposed for utilizing motions in Fluent UI React components?
```

Comment:
Do you plan to restrict developers to only be able to override motions using this API? Can users override the default motions using plain CSS for e.g.?